server/indexer/indexer_config.go

```diff
@@ -0,0 +1,37 @@
+package indexer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+)
+
+type Index struct {
+	Kind string `json:"kind"`
+}
+
+type IndexesConfig struct {
```

Comment:
Better naming conventions:

```suggestion
type Config struct {
```

`Config` is enough, as users of this package requires writing `indexer.Config{}`, but it is easily understandable what config it is.

In general, if the package name is `indexer`:

```go
package indexer

type Indexer struct {} // it's ok
type IndexerHogehoge struct {} // -> Hogehoge

func NewIndexer() *Indexer // -> New
func NewIndexerHogehoge() *IndexerHogehoge // -> NewHogehoge
```